### PR TITLE
ci: update README to include e2e pipeline

### PR DESCRIPTION
**Pull request by Yuggupta27**
The README is updated with e2e jenkins pipeline
information.

Signed-off-by: Yug Gupta <ygupta@redhat.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,10 +35,36 @@ following files:
...
 
+- `e2e.groovy` is the Jenkins Pipeline responsible for running e2e tests on
+  a multi-node kubernetes cluster hosted on [Centos CI](https://wiki.centos.org/QaWiki/CI).
+  It verifies complete e2e functionalities for a corresponding Pull request
+  on [ceph-csi](https://github.com/ceph/ceph-csi/tree/master) repository.
+  It executes the following stages:
+
+  1. dynamically allocate a Jenkins Slave (`node('cico-workspace')`) with tools
+     and configuration to request a bare-metal machine.
+  2. checkout the `centos/ci` branch of the repository, which contains scripts
```
Comment (nixpanic) [inline README.md:46]: you can just number everything with `1.` and markdown will number them correctly

File: README.md [hunk 2]
```diff
@@ -35,10 +35,36 @@ following files:
   1. as final step, return the bare-metal machine to the CentOS CI for other
      users (it will be re-installed with a minimal CentOS environment again)
 
+- `e2e.groovy` is the Jenkins Pipeline responsible for running e2e tests on
+  a multi-node kubernetes cluster hosted on [Centos CI](https://wiki.centos.org/QaWiki/CI).
+  It verifies complete e2e functionalities for a corresponding Pull request
+  on [ceph-csi](https://github.com/ceph/ceph-csi/tree/master) repository.
```
Comment (nixpanic) [inline README.md:41]: `/tree/master` is not needed, it is the default location for the repository

File: README.md [hunk 3]
```diff
@@ -35,10 +35,36 @@ following files:
   1. as final step, return the bare-metal machine to the CentOS CI for other
      users (it will be re-installed with a minimal CentOS environment again)
 
+- `e2e.groovy` is the Jenkins Pipeline responsible for running e2e tests on
```
Comment (nixpanic) [inline README.md:38]: maybe add that "e2e" means "End to End"

File: README.md [hunk 4]
```diff
@@ -35,10 +35,36 @@ following files:
...
+     for provisioning and preparing the environment for running tests.
+  3. reserve a bare-metal machine with `cico` (configured on the Jenkins Slave);
+     retry if not immediately available.
+  4. provision the reserved bare-metal machine with additional tools and
+     dependencies to run the test (see `prepare.sh` below).
+  5. set up a multi-node k8s cluster on the bare-metal machine for performing
+     e2e tests (see `multi-node-k8s.sh` below).
+  6. deploy rook on the multi node kubernetes cluster.
+  7. run the e2e tests on the configured setup against the corresponding
+     pull request on the [ceph-csi](https://github.com/ceph/ceph-csi/tree/master)
```
Comment (nixpanic) [inline README.md:56]: shorten the url here too, or even better: use labels and list the label+url at the bottom of the file

Comment (Yuggupta27) [reply README.md:56]: Thanks, I'll make the changes

**Comment by nixpanic** [comment]
@mergifyio rebase

File: README.md [hunk 5]
```diff
@@ -35,10 +35,36 @@ following files:
   1. as final step, return the bare-metal machine to the CentOS CI for other
      users (it will be re-installed with a minimal CentOS environment again)
 
+- `e2e.groovy` is the Jenkins Pipeline responsible for running End-to-End tests on
+  a multi-node kubernetes cluster hosted on [Centos CI](centos_ci).
```
Comment (nixpanic) [inline README.md:39]: Note that the format is `[title][id]` and the `[id]` needs to be available at the bottom of the document.

**Comment by nixpanic** [comment]
(FYI, I just asked mergify to rebase it so the jobs for the ci/centos bransh get started)

**Comment by nixpanic** [comment]
@Mergifyio rebase

**Comment by nixpanic** [comment]
The failing of `ci/centos/jjb-validate` has been reported in #1111

**Comment by Madhu-1** [comment]
@Yuggupta27 can you please revisit this one?

File: README.md [hunk 6]
```diff
@@ -35,10 +35,36 @@ following files:
...
+  a multi-node kubernetes cluster hosted on [Centos CI][centos_ci].
+  It verifies complete e2e functionalities for a corresponding Pull request
+  on [ceph-csi][git_repo] repository.
+  It executes the following stages:
+
+  1. dynamically allocate a Jenkins Slave (`node('cico-workspace')`) with tools
+     and configuration to request a bare-metal machine.
+  1. checkout the `centos/ci` branch of the repository, which contains scripts
+     for provisioning and preparing the environment for running tests.
+  1. reserve a bare-metal machine with `cico` (configured on the Jenkins Slave);
```
Comment (yati1998) [inline README.md:48]: reserve -> reserves.

File: README.md [hunk 7]
```diff
@@ -35,10 +35,36 @@ following files:
...
+     pull request on the [ceph-csi][git_repo]
+     repository.
+  1. as final step, return the bare-metal machine to the CentOS CI for other
+     users (it will be re-installed with a minimal CentOS environment again).
+
 - `prepare.sh` installs dependencies for the test, and checks out the git
   repository and branch (or Pull Request) that contains the commits to be
   tested (and the test itself)
 
+- `multi-node-k8s.sh` installs the dependencies, and sets up a multi-node
```
Comment (yati1998) [inline README.md:65]: ',' after dependencies can be removed.

**Pull request closed without merging**